openfermioncirq/variational/ansatzes/low_rank.py

```diff
@@ -0,0 +1,303 @@
+        """Produce the operations of the ansatz circuit."""
+
+        n_qubits = len(qubits)
+        param_set = set(self.params())
+
+        for i in range(self.iterations):
+
+            # Simulate the off-diagonal one-body terms.
+            def one_body_interaction(p, q, a, b) -> cirq.OP_TREE:
+                t_symbol = LetterWithSubscripts('T', p, q, i)
+                if t_symbol in param_set:
+                    yield XXYYGate(half_turns=t_symbol).on(a, b)
+            yield swap_network(qubits, one_body_interaction, fermionic=True)
+            qubits = qubits[::-1]
+
+            # Simulate the diagonal one-body terms.
+            for p in range(n_qubits):
+                u_symbol = LetterWithSubscripts('U', p, i)
+                if u_symbol in param_set:
+                    yield cirq.RotZGate(half_turns=u_symbol).on(qubits[p])
+
+            # Simulate each singular vector of the two-body terms.
+            prior_basis_matrix = numpy.identity(n_qubits)
+
+            for j in range(len(self.eigenvalues)):
+
+                # Get the basis change matrix.
+                basis_change_matrix = self.basis_change_matrices[j]
+
+                # Perform basis change.
+                inverse_basis_matrix = basis_change_matrix.T.conj()
+                merged_basis_matrix = numpy.dot(prior_basis_matrix,
+                                                inverse_basis_matrix)
+                yield bogoliubov_transform(qubits, merged_basis_matrix)
+
+                # Simulate the off-diagonal two-body terms.
+                def two_body_interaction(p, q, a, b) -> cirq.OP_TREE:
+                    v_symbol = LetterWithSubscripts('V', p, q, j, i)
+                    if v_symbol in param_set:
+                        yield cirq.Rot11Gate(half_turns=v_symbol).on(a, b)
+                yield swap_network(qubits, two_body_interaction)
+                qubits = qubits[::-1]
+
+                # Simulate the diagonal two-body terms.
+                for p in range(n_qubits):
+                    u_symbol = LetterWithSubscripts('U', p, j, i)
+                    if u_symbol in param_set:
+                        yield cirq.RotZGate(half_turns=u_symbol).on(qubits[p])
+
+                # Update prior basis change matrix.
+                prior_basis_matrix = basis_change_matrix
+
+            # Undo final basis transformation.
+            yield bogoliubov_transform(qubits, prior_basis_matrix)
+
+        if not self.omit_final_swaps:
+            # If the number of fermionic swap networks was odd,
+            # swap the modes back
+            if self.iterations & 1:
+                yield swap_network(qubits, fermionic=True)
+                # If the total number of swap networks was odd,
+                # swap the qubits back
+                if len(self.eigenvalues) & 1:
+                    yield swap_network(qubits)
+
+
+    def default_initial_params(self) -> numpy.ndarray:
```

Comment:
I'm fine merging things for now, but in general isn't this function something that could be made more general to work for a variety of Trotter based variational algorithms?

Reply:
Yes, I suppose so. Let's leave generalizing this for another PR.

Reply:
that's fine